theano/tensor/nnet/ctc.py

```diff
@@ -175,13 +176,16 @@ def make_node(self, activations, labels, input_lengths):
         if t_input_lengths.ndim != 1:
             raise ValueError('input_lengths must have 1 dimension.')
 
+        openmp_enabled = 1 if gof.OpenMPOp(self).openmp is not None else 0
+        openmp_enabled = as_scalar(openmp_enabled).astype('int32')
+
         costs = T.fvector(name="ctc_cost")
         outputs = [costs]
         if self.compute_grad:
             gradients = T.ftensor3(name="ctc_grad")
             outputs += [gradients]
 
-        return gof.Apply(self, inputs=[t_activations, t_labels, t_input_lengths],
+        return gof.Apply(self, inputs=[t_activations, t_labels, t_input_lengths, openmp_enabled],
```

Comment:
I don't see a problem, but I'm not sure this is good.

@abergeron what do you think of this?

Should this be a param?

Ideally, we should allow this to change when we pickle/unpickle, but we didn't supported that up to now. So this isn't a requirement.